Source/ActionController.swift

```diff
@@ -262,39 +274,38 @@ open class ActionController<ActionViewType: UICollectionViewCell, ActionDataType
             let layoutAtts = collectionViewLayout.layoutAttributesForItem(at: IndexPath(item: section.actions.count - 1, section: hasHeader() ? lastSectionIndex + 1 : lastSectionIndex))
             contentHeight = layoutAtts!.frame.origin.y + layoutAtts!.frame.size.height
 
-            if settings.cancelView.showCancel && !settings.cancelView.hideCollectionViewBehindCancelView {
-                contentHeight += settings.cancelView.height
+            if settings.cancelView.hasCancelView, !settings.cancelView.displayInFooter {
+                if let cancel = cancelView, cancelView != nil {
+                    contentHeight += cancel.frame.height
+                }
             }
         }
-
         setUpContentInsetForHeight(view.frame.height)
         
         // set up collection view initial position taking into account top content inset
         collectionView.frame = view.bounds
-        collectionView.frame.origin.y += contentHeight + (settings.cancelView.showCancel ? settings.cancelView.height : 0)
+        collectionView.frame.origin.y += contentHeight
+        
+        if let cancel = cancelView, !settings.cancelView.displayInFooter {
+            collectionView.frame.origin.y += contentHeight + (settings.cancelView.hasCancelView ? cancel.frame.height : 0)
+        }
         collectionViewLayout.footerReferenceSize = CGSize(width: 320, height: 0)
         // -
         
-        if settings.cancelView.showCancel {
-            if cancelView == nil {
+        if settings.cancelView.hasCancelView, !settings.cancelView.displayInFooter {
+            if cancelView != nil {
```

Comment:
I'm not sure why this is needed. Looks like if the cancelView exists, then you are adding a button on top of it? I guess that this is done in order to hide the controller when the cancel view is "tapped", is it right?

In such case, I think that a better approach would be adding a constraint to the cancelView to force it to conform some protocol. What do you think?